libs/types/src/cdf/ballot-definition/convert.ts

```diff
@@ -30,6 +31,326 @@ function officeId(contestId: Vxf.ContestId): string {
+): string | undefined {
+  return (ballotDefinition.Office ?? []).find(
+    (office) => office['@id'] === officeId(contestId)
+  )?.Term.Label;
+}
+
+const SUPPORTED_LANGUAGES = new Set<string>(Object.values(LanguageCode));
+
+/**
+ * String translation string key, with support for one level of optional nesting
+ * for strings of the same type that vary based on content ID(e.g.
+ * `['contestTitle', contest.id]`).
+ *
+ * See https://www.i18next.com/translation-function/essentials#accessing-keys
+ */
+type StringKey = ElectionStringKey | [ElectionStringKey, string];
+
+/**
+ * Sets the appropriate language strings in supported languages for the given
+ * internationalized CDF ballot content text.
+ */
+function setInternationalizedUiStrings(params: {
+  uiStrings: UiStringsPackage;
+  stringKey: StringKey;
+  values: readonly Cdf.LanguageString[];
+}) {
+  const { stringKey, uiStrings, values } = params;
+
+  for (const value of values) {
+    const languageCode = value.Language;
+    if (!SUPPORTED_LANGUAGES.has(languageCode)) {
+      continue;
+    }
+
+    const valuePath = [languageCode, stringKey].flat();
+    setWith(uiStrings, valuePath, value.Content, Object);
+  }
+}
+
+/**
+ * Sets the default English language string for the given ballot content text.
+ * Used for content that will be spoken, but not translated, like ballot style.
+ */
+function setStaticUiString(params: {
+  uiStrings: UiStringsPackage;
+  stringKey: StringKey;
+  value: string;
+}) {
+  const { stringKey, uiStrings, value } = params;
+
+  const valuePath = [LanguageCode.ENGLISH, stringKey].flat();
+  setWith(uiStrings, valuePath, value, Object);
+}
+
+/**
+ * Returns all the language codes in use in a CDF election based on the election
+ * title internationalized text.
+ */
+function electionLanguageCodes(cdfElection: Cdf.BallotDefinition): string[] {
+  const electionTitleStrings = assertDefined(cdfElection.Election[0]).Name.Text;
+  return electionTitleStrings
+    .map((string) => string.Language)
+    .filter((languageCode) => SUPPORTED_LANGUAGES.has(languageCode));
+}
+
+const extractorFns: Record<
+  ElectionStringKey,
+  (cdfElection: Cdf.BallotDefinition, uiStrings: UiStringsPackage) => void
+> = {
+  [ElectionStringKey.BALLOT_LANGUAGE](cdfElection, uiStrings) {
+    // CDF does not support internationalized text for this string, so we just
+    // use JS to internationalize the language code.
+    setInternationalizedUiStrings({
+      uiStrings,
+      stringKey: ElectionStringKey.BALLOT_LANGUAGE,
+      values: electionLanguageCodes(cdfElection).map(
+        (languageCode): Cdf.LanguageString => ({
+          '@type': 'BallotDefinition.LanguageString',
+          Language: languageCode,
+          Content: assertDefined(
+            new Intl.DisplayNames([languageCode], {
+              type: 'language',
+              style: 'narrow',
+              fallback: 'none',
+            }).of(languageCode)
+          ),
+        })
+      ),
+    });
+  },
```

Comment:
We previously didn't do anything in the extractor function for ballot language, since we could rely on electionStrings.json to provide the translation. Now, if we get a CDF election, we won't have an additional electionStrings.json, so we have to make do and fill in a reasonable default here. I emulated the approach taken in VxDesign's translation code to translate ballot language.